core/node/node_framework/src/service/tests.rs

```diff
@@ -0,0 +1,235 @@
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use futures::lock::Mutex;
+use tokio::runtime::Runtime;
+
+use crate::service::{
+    ServiceContext, StopReceiver, Task, WiringError, WiringLayer, ZkStackService,
+};
+
+// ZkStack Service's `new()` method has to have a check for nested runtime.
+#[test]
+fn test_new_with_nested_runtime() {
+    let runtime = Runtime::new().unwrap();
+
+    runtime.block_on(async {
+        assert_eq!(
+            ZkStackService::new().unwrap_err().to_string(),
+            "Detected a Tokio Runtime. ZkStackService manages its own runtime and does not support nested runtimes".to_string()
+        );
+    })
+}
+
+#[derive(Debug)]
+struct DefaultLayer;
+
+#[async_trait::async_trait]
+impl WiringLayer for DefaultLayer {
+    fn layer_name(&self) -> &'static str {
+        "default_layer"
+    }
+
+    async fn wire(self: Box<Self>, mut _node: ServiceContext<'_>) -> Result<(), WiringError> {
+        Ok(())
+    }
+}
+
+// ZkStack Service's `add_layer()` method has to add multiple layers into `self.layers`.
+#[test]
+fn test_add_layer() {
+    let mut zk_stack_service = ZkStackService::new().unwrap();
+    zk_stack_service
+        .add_layer(DefaultLayer)
+        .add_layer(DefaultLayer);
+    assert_eq!(2, zk_stack_service.layers.len());
```

Comment:
It's almost always a good idea to provide a message for assertions to make it easier to debug if some of tests fail. E.g. here the default message would be something like "Assertion failed, expected 2 but got 1". If you add a message like "Incorrect number of layers in the service", it'll be much more helpful.